File: House Robber 2/Recursion.py
def HouseRobber(size,arr):
    if size == 0: return arr[0]
    if size < 0: return 0

    pick = arr[size] + HouseRobber(size-2,arr)
    nonPick = HouseRobber(size-1,arr)

    return max(pick,nonPick)

if __name__ == '__main__':
    arr = [1,2,3,1]
    size =  len(arr)

    arr1 = arr[:size-1]
    arr2 = arr[1:]

    print( max(HouseRobber(size-2,arr1) , HouseRobber(size-2,arr2)) )

File: House Robber 2/Recursion.java

class Recursion {
     public static int maxRecursion(int size,int arr[]) {
        if(size < 0) return 0;
        if(size == 0) return arr[size];

        int pick = arr[size] + maxRecursion(size-2,arr);
        int nonPick = maxRecursion(size-1,arr);

        return Math.max(pick,nonPick);
    }

    public static void main(String[] args) {
        int[] nums = {1,2,3,1};
        int size = nums.length;   

        int arr1[] = new int[size-1];
        int arr2[] = new int[size-1];

        int k = 0;
        for(int i = 0 ; i < size; i ++ ) {
            if( i > 0) {
                arr2[k] = nums[i];
                k += 1;
            }

            if( i < size-1) arr1[i] = nums[i];
        }

        System.out.println(Math.max( maxRecursion(size-2,arr1) , maxRecursion(size-2,arr2)) );

    }
}